vsintegration/src/FSharp.Editor/CodeLens/CodeLens.fs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace rec Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Windows.Controls
+open System.Windows.Media
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Editor
+open Microsoft.VisualStudio.Text.Formatting
+open System.ComponentModel.Composition
+open Microsoft.VisualStudio.Utilities
+open Microsoft.CodeAnalysis
+
+type internal CodeLensAdornment(view:IWpfTextView) as this=
```

Comment:
don't actually think this is the right approach - the VS Platform defines an adornment already, but it needs a certain type of tag to know where to draw it.

The code for that tagging for C#/VB is all closed source unfortunately.

@olegtk do you know if the tag types, etc are exposed publicly for another language to add CodeLens support?